de.metas.contracts/src/main/java/de/metas/contracts/subscription/impl/SubscriptionService.java

```diff
@@ -89,16 +86,23 @@ public static int changeRecipient(@NonNull final ChangeRecipientsRequest changeR
 		@Builder
 		private ChangeRecipientsRequest(
 				@NonNull final I_C_Flatrate_Term term,
-				@NonNull final Timestamp dateFrom,
-				@NonNull final Timestamp dateTo,
+				final Timestamp dateFrom,
+				final Timestamp dateTo,
 				final int DropShip_BPartner_ID,
 				final int DropShip_Location_ID,
 				final int DropShip_User_ID,
 				boolean IsPermanentRecipient)
 		{
 			this.term = term;
 
-			Preconditions.checkArgument(!dateTo.before(dateFrom), "The given dateTo may not be before the given dateFrom; dateFrom=%s; dateTo=%s", dateFrom, dateTo);
+			this.IsPermanentRecipient = IsPermanentRecipient;
+
+			if (!IsPermanentRecipient)
+			{
+				Preconditions.checkArgument(!dateTo.before(dateFrom), "The given dateTo may not be before the given dateFrom; dateFrom=%s; dateTo=%s", dateFrom, dateTo);
+				Preconditions.checkNotNull(dateFrom, "The dateForm shall not be null");
```

Comment:
better do the null-check before the before-check :-)

Reply:
pfu.. right! :)))